src/simulate.jl

```diff
@@ -1,8 +1,46 @@
+"""
+See [`simulate!`](@ref)
+"""
+function simulate end
+
+function simulate(rng::AbstractRNG, m::MixedModel{T}, newX::AbstractMatrix = m.X;
+                  kwargs...) where {T}
+    size(newX, 1) == nobs(m) ||
+        throw(DimensionMismatch("New fixed-effect model matrix must have the same number of observations as the original."))
+    size(newX, 2) == size(m.X, 2) ||
+    throw(DimensionMismatch("New fixed-effect model matrix must have the same predictors as the original."))
+    y = zeros(T, nobs(m))
+    simulate!(rng, y, m, newX; kwargs...)
```

Comment:
I think the next line, `y`, is redundant as `simulate!` returns the modified `y` argument.